src/pkg/cli/client/byoc/aws/domain.go

```diff
@@ -54,44 +57,58 @@ func prepareDomainDelegation(ctx context.Context, projectDomain string, r53Clien
 	return delegationSet.NameServers, delegationSetId, nil
 }
 
-func getOrCreateDelegationSet(ctx context.Context, r53Client aws.Route53API) (*types.DelegationSet, error) {
-	// Avoid creating a new delegation set if one already exists
-	delegationSet, err := aws.GetDelegationSet(ctx, r53Client)
-	// Create a new delegation set if it doesn't exist
-	if errors.Is(err, aws.ErrNoDelegationSetFound) {
-		// Create a new delegation set. There's a race condition here, where two deployments could create two different delegation sets,
-		// but this is acceptable because the next time the zone is deployed, we'll get the existing delegation set from the zone.
-		delegationSet, err = aws.CreateDelegationSet(ctx, nil, r53Client)
-	}
-	if err != nil {
-		return nil, err
+func createUsableDelegationSet(ctx context.Context, domain string, r53Client aws.Route53API, resolverAt func(string) dns.Resolver) (*types.DelegationSet, error) {
+	// Try up to 5 times to create a delegation set that is usable (i.e., none of its NS servers have conflicting records for the domain)
```

Comment:
It might be helpful to provide a little background context here, but I'm not actually sure I can explain why we need to find a delegation set without NS records for our domain. I think it has something to do with stacks, but I can't clearly lay it out. Would you mind filling this out?
```suggestion
	// route53 assigns a random selection of name servers when creating a
	// delegation set. If we get a delegation set which contains a name server
	// which already has an NS record for this hosted zone, we cannot use it
	// because [...] @edwardrf help describe this more clearly
	// Try up to 5 times to create a delegation set that is usable (i.e., none
	// of its NS servers have conflicting records for the domain)
```